homeassistant/components/ps4/media_player.py

```diff
@@ -272,12 +275,34 @@ def get_device_info(self, status):
             'manufacturer': 'Sony Interactive Entertainment Inc.',
             'sw_version': sw_version
         }
-        self._unique_id = status['host-id']
+
+        # Use last 4 Chars of credential as suffix. Unique ID per PSN user.
+        suffix = self._creds[-4:]
+        self._unique_id = "{}_{}".format(status['host-id'], suffix)
+
+    async def async_update_entity_data(self):
+        """From 0.89. Update identifier. Prevent changing entity_id."""
+        registry = await entity_registry.async_get_registry(self.hass)
+        unique_id = self._unique_id.split('_')
+        unique_id = unique_id[0]
+        entity_id = registry.async_get_entity_id(
+            'media_player', PS4_DOMAIN, unique_id)
+
+        # Remove old entity entry. Update current entry with old entity_id.
+        if entity_id is not None:
+            registry.async_remove(entity_id)
+            old_entity_id = registry.async_get_entity_id(
```

Comment:
I think we should call this `current_entity_id`. `old_entity_id` here is confusing, since that's what we should rename `entity_id` to.